scipy/stats/stats.py

```diff
@@ -2306,6 +2311,186 @@ def zmap(scores, compare, axis=0, ddof=0):
+    ranges than the actual IQR. For example, setting ``range=(0, 100)`` is
+    equivalent to ``numpy.ptp``.
+
+    The IQR of an empty array is scalar zero.
+
+    .. versionadded:: 0.18.0
+
+    Parameters
+    ----------
+    x : array_like
+        Input array or object that can be converted to an array.
+    axis : int or sequence of int, optional
+        Axis along which the range is computed. The default is to compute the
+        IQR for the entire array.
+    range : Two-element sequence containing floats in range of [0,100]
+        Percentiles over which to compute the range. Each must be between 0 and
+        100, inclusive. The default is the true IQR: (25, 75). The order of the
+        elements is not important.
+    scale : scalar or str
+        The numerical value of scale will be divided out of the final result.
+        The following string values are recognized:
+
+          'raw' : No scaling, just return the raw IQR.
+          'normal' : Scale by :math:`2 \\sqrt{2} erf^{-1}(\\frac{1}{2})`, which
+              is approximately `1.349`.
+
+        The default is 'raw'. Array-like scale is also allowed, as long as it
+        broadcasts correctly to the output such that ``out / scale`` is a valid
+        operation. The output dimensions depend on the input array, `x`, the
+        `axis` argument, and the `keepdims` flag.
+    nan_policy : {'propagate', 'raise', 'omit'}, optional
+        Defines how to handle when input contains nan. 'propagate' returns nan,
+        'raise' throws an error, 'omit' performs the calculations ignoring nan
+        values. Default is 'propagate'.
+    interpolation : {'linear', 'lower', 'higher', 'midpoint', 'nearest'}
+        This optional parameter specifies the interpolation method to use
+        when the percentile boundaries lie between two data points `i` and `j`:
+            * linear: `i + (j - i) * fraction`, where `fraction` is the
+              fractional part of the index surrounded by `i` and `j`.
+            * lower: `i`.
+            * higher: `j`.
+            * nearest: `i` or `j` whichever is nearest.
+            * midpoint: (`i` + `j`) / 2.
+    keepdims : bool, optional
+        If this is set to `True`, the reduced axes are left in the result as
+        dimensions with size one. With this option, the result will broadcast
+        correctly against the original array `x`.
+
+    Returns
+    -------
+    iqr : scalar or ndarray
+        If ``axis=None``, a scalar is returned. If the input contains integers
+        or floats of smaller precision than 64, then the output data-type is
+        ``np.float64``. Otherwise, the output data-type is the same as that of
+        the input.
+
+    References
+    ----------
+    .. [1] "Interquatrile range" https://en.wikipedia.org/wiki/Interquartile_range
+    .. [2] "Interquatrile range" http://www.stattrek.com/statistics/dictionary.aspx?definition=Interquartile%20range
```

Comment:
do we need the second link? It is a not too popular website, chances are that the link is broken a few years from now